stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -32,37 +165,23 @@ namespace math {
  *   x is nan or infinite
  */
 template <typename T_x, typename T_sigma, typename T_l>
-inline
-    typename Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
-                           Eigen::Dynamic, Eigen::Dynamic>
-    gp_exp_quad_cov(const std::vector<T_x> &x, const T_sigma &sigma,
-                    const T_l &length_scale) {
-  using std::exp;
+inline typename Eigen::Matrix<return_type_t<T_x, T_sigma, T_l>, -1, -1>
```

Comment:
While it is more verbose, I prefer `Eigen::Dynamic` to just `-1`. It makes the meaning of code clearer. Also I don't think Eigen guarantees that That `Dynamic` will be represented by `-1`. They could change that in future versions.

Reply:
> While it is more verbose, I prefer Eigen::Dynamic to just -1. It makes the meaning of code clearer. Also I don't think Eigen guarantees that That Dynamic will be represented by -1. They could change that in future versions.

I agree with the first point but second is a little ehhh. But yeah we can switch these back. I went a little Marie Kondo when cleaning the templates up